Roles/Impostor/DoubleAgent.cs

```diff
@@ -0,0 +1,358 @@
+                if ((!NameNotifyManager.Notice.TryGetValue(pc.PlayerId, out var a) || a.Item1 != Duration) && Duration != string.Empty) pc.Notify(Duration, 1.1f);
+            }
+        }
+    }
+
+    // Players go bye bye ¯\_(ツ)_/¯
+    private void BoomBoom(PlayerControl player)
+    {
+        if (player.inVent) player.MyPhysics.RpcBootFromVent(player.GetPlayerVentId());
+
+        foreach (PlayerControl target in Main.AllAlivePlayerControls) // Get players in radius of bomb that are not in a vent.
+        {
+            if (CheckForPlayersInRadius(player, target) <= ExplosionRadius.GetFloat())
+            {
+                if (player.inVent) return;
+                Main.PlayerStates[target.PlayerId].deathReason = PlayerState.DeathReason.Bombed;
+                target.RpcMurderPlayer(target);
+            }
+        }
+
+        PlaySoundForAll("Boom");
+        ClearBomb();
+
+        _Player.Notify(Utils.ColorString(Utils.GetRoleColor(CustomRoles.DoubleAgent), GetString("DoubleAgent_BombExploded")));
+    }
+
+    private static float CheckForPlayersInRadius(PlayerControl player, PlayerControl target) => Vector2.Distance(player.GetCustomPosition(), target.GetCustomPosition());
+
+    // Play specific sound for all players.
+    private static void PlaySoundForAll(string Sound)
+    {
+        foreach (PlayerControl player in Main.AllPlayerControls)
+        {
+            player.RPCPlayCustomSound(Sound);
+        }
+    }
+
+    // Set bomb mark on player.
+    public override string GetMark(PlayerControl seer, PlayerControl seen = null, bool isForMeeting = false)
+    {
+        if (seen == null ) return string.Empty;
+        if (CurrentBombedPlayers.Contains(seen.PlayerId)) return Utils.ColorString(Color.red, "Ⓑ"); // L Rizz :)
+        return string.Empty;
+    }
+
+
+    // Set timer for Double Agent Modded Clients.
+    public override string GetLowerText(PlayerControl player, PlayerControl seen = null, bool isForMeeting = false, bool isForHud = false)
+    {
+        if (player == null) return string.Empty;
+        if (CurrentBombedTime > 0 && CurrentBombedTime < BombExplosionTimer.GetFloat() + 1) return Utils.ColorString(player.GetRoleColor(), string.Format(GetString("DoubleAgent_BombExplodesIn"), (int)CurrentBombedTime));
+        return string.Empty;
+    }
+
+    // Send bomb timer to Modded Clients when active.
+    private void SendRPC()
+    {
+        var writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId, (byte)CustomRPC.SyncRoleSkill, SendOption.None, -1);
+        writer.WriteNetObject(_Player);
+        writer.WritePacked((int)CurrentBombedTime);
+        AmongUsClient.Instance.FinishRpcImmediately(writer);
+    }
+
+    // Receive and set bomb timer from Host when active.
+    public override void ReceiveRPC(MessageReader reader, PlayerControl NaN)
+    {
+        CurrentBombedTime = reader.ReadInt32();

```

Comment:
Since you are passing WritePacked you need to accept this as `reader.ReadPackedInt32()`